Suggested changes for Delta height workaround

thinkyhead opened the pull request:
Some suggested changes.

- Rearrange `probe_pt` args for shorter calls.
- Conditions for use: `ENABLED(DELTA_AUTO_CALIBRATION) && HAS_MESH`

Bob-the-Kuhn committed:
snapshot

found a few more nasties

Bob-the-Kuhn committed:
as good as i can get it

minor edits to force new Travis test (last one timed out)

Update u8g_com_HAL_DUE_st7920_sw_spi.cpp

initial power up display garbage gone

Bob-the-Kuhn committed:
Merge pull request #9028 from Bob-the-Kuhn/Due-SDSUPPORT

[2.0.x] Due - allow compile without SD card enabled & misc improvements (replaces PR #9007)

thinkyhead committed:
M303 pid autotune cleanup, commentary

thinkyhead committed:
Apply FILAMENT_LOAD_UNLOAD_GCODES to example configs

thinkyhead committed:
Group leveling reports for folding

thinkyhead committed:
Improvements to LCD menu macros

thinkyhead committed:
Nozzle park args

thinkyhead committed:
Merge pull request #9040 from thinkyhead/bf2_m303_bughunt

[2.0.x] M303 pid autotune cleanup, comment

studiodyne committed:
2.0  Fix for bad Pause extruder position resuming (#9029)

Bob-the-Kuhn committed:
Merge pull request #9009 from Bob-the-Kuhn/Due-RRDFG-power-up-garbage

[2.0.x] Due - fix power up garbage on RRDFG LCD

revilor committed:
Fix for issue #9005, #8942 - Bed PID autotuning (#9036)

thinkyhead committed:
M701 / M702 language changes

thinkyhead committed:
Add M701/M702 Filament Load/Unload, M603

thinkyhead committed:
M702.cpp => M701_M702.cpp

Roxy-3D committed:
Add enough precision so planner.xy_skew_factor doesn't print as 0.00

Roxy-3D committed:
Add precision to planner.xy_skew_factor so it doesn't print as 0.00

Roxy-3D committed:
use SERIAL_EOL(); instead of new line string

thinkyhead committed:
CheapTRONIC v2.0 additions/revisions

Based on #9030

thinkyhead committed:
Merge pull request #9042 from thinkyhead/bf2_M701_M702

[2.0.x] Improved Pause/Load/Unload, M701-M702

thinkyhead committed:
Pretty up TMC current write

thinkyhead committed:
Add MarlinSettings::validate()

thinkyhead committed:
Add a struct defining EEPROM storage

thinkyhead committed:
Add data size validation

thinkyhead committed:
Write filler with NO_VOLUMETRICS

thinkyhead committed:
Add FIELD_TEST to validate storage

thinkyhead committed:
UBL G29 works without settings.load()

thinkyhead committed:
Trailing whitespace clean

thinkyhead committed:
Detab some HAL files

thinkyhead committed:
Merge pull request #9056 from thinkyhead/bf2_new_eeprom_powers

[2.0.x] New EEPROM powers

thinkyhead committed:
Settings FIELD_TEST fix

Bob-the-Kuhn committed:
1234

Bob-the-Kuhn committed:
Merge pull request #9068 from Bob-the-Kuhn/SD-detect

[2.0.x] SD card detect improvement

Bob-the-Kuhn committed:
works

root cause identified

GMagician committed:
Bmp fix

Clockwise arrow fixed

thinkyhead committed:
Merge pull request #9077 from GMagician/2.0.x-bmp-fix

[2.0.x] Clockwise bmp fix

thinkyhead committed:
Laser/spindle PWM off on M5

Bob-the-Kuhn committed:
Merge pull request #9069 from Bob-the-Kuhn/broken-teensy

[2.0.x] fix Teensy2 compile error (PlatformIO LDF problem)

thinkyhead committed:
Move MBL functions into the class

thinkyhead committed:
Fix M503 skew output

Followup to 19e75ccf8618b3691f14f6d51b1e73f56fb1355e

Commit:
Set has_mesh if user issues G9 S3

Prevents user from having to manually run through entire manual leveling process only to enable mesh so they can manually restore their offline mesh backup

thinkyhead committed:
Merge pull request #9086 from fiveangle/bf2-hasmesh_on_g29s3

[2.0.x] Set has_mesh if user issues G9 S3 with MBL

thinkyhead committed:
Make mbl.has_mesh() a method

tcm0116 commented:
This has become extremely frustrating. I'm trying to find a legitimate solution to a real problem. You seem to be so caught up on delta height and babystepping that I think you genuinely don't understand the actual problem that I'm trying to solve. 

I'm not trying to find a backdoor way to fixing errors with delta height. I'm trying to find a way to provide consistent operation of live z height adjustment for delta machines that provides consistent behavior to cartesian machines. 

Furthermore, this problem isn't limited to delta machines. Any machine that uses the probe for leveling but not for homing will have this same problem.

Machines that use the probe for homing and leveling have the benefit of being able to apply changes to the z offset made during printing the first layer when the next homing event occurs because the z offset is included as part of the homing process. Machines that don't use the probe for homing have no choice but to re-execute G29 in order to incorporate the change to the z offset. As such, a large portion of the benefit of UBL is lost some they now have to run G29 if their z offset wasn't exactly correct. 

I think you're getting hung up on the fact that the z offset ends up correcting for both the actual offset between the nozzle and the probe as well as any error in nozzle's position relative to the bed (i.e. error in delta height or error in the homing position for non-deltas that don't home with the probe). I understand that you want to keep them 100% unrelated, but when you use the probe for leveling but not for homing, they become related because the probed position is a function of logical nozzle position and the z offset.

My argument is this: from the user's perspective, what does it really matter if a small amount of error in the logical position translates over to the z offset? The user wants the ability to get consistent results, which we currently do not provide for machines that use a probe for leveling but not homing. 

I'm truly trying to make Marlin better because I think it's a great project. However, your combative attitude with this particular issue has been a big turn off. I'll not trying to sneak some half baked hack though at the direction of another person. I'm trying to provide a solution to a real problem. If you see holes in it, that's fine: I'll try to fix them, but please don't just assume that I'm trying to spite you.

thinkyhead committed:
Merge pull request #9087 from thinkyhead/bf2_mbl_localize

[2.0.x] MBL cleanup, has_mesh() method

Tannoo committed:
[2.0.x] UBL Mesh Storage Message (#9095)

thinkyhead committed:
Fix TMC current read from EEPROM

GMagician committed:
[2.0.x] add semitec 104NT-4 (#9119)

Table is very very similar to old 104-GT2 so I added only the name and recycled old ID

GMagician committed:
2.0.x Fix compiler warning (#9116)

Fix duplicate language define warning

thinkyhead committed:
More EEPROM field debugging

thinkyhead commented:
I never assumed that.

If this is taking too long to resolve, I apologize. But the flood of reports about "high on one side and low on the other," needed to be resolved with some finality. I had over 1000 unread issues to deal with and needed to clear those to make room for new and interesting developments.

I'm not simply trying to preserve the _status quo_, and I'm not being intentionally obtuse, and I'm certainly not being "hostile." I have questions, and I sense implications, but I have lacked the space and time to delve into the thing. It is important for me to understand the code, and while I was without a delta of my own to test with I could only indirectly see its effects in action.

You have been patient, but certain others have not been, and I don't need extra pressure and drumbeats on top of the existing pressure. These changes will get in, in one form or another. In the meantime what we have now is much the same as it's been, and no one will fault us for that, given the countless other things we've fixed in the interim.

I created the Slack to see if it would facilitate collaboration in a less stagey forum. It has done a little to help with back-and-forth exchange, and I like the conversations we've had around these things. It's important for me to see through the eyes of others, and the daily grind just fixing the actual bugs people report has made it difficult to give deeper attention to the matter.

I can delegate to some extent, but the handful of times I've stepped back and let faith take the lead it quickly turned into a weekend in Vegas. Just getting the leveling system to the point where it wasn't a convoluted mess took many RC releases and many long and often heated discussions. Now, if I'm not looking at it every day, all knowledge about it flies out of my head.

Now that we've established that twisted gantries and angled probes have been the real culprits in so many "there's definitely a bug!" reports, I'm feeling more relaxed about the leveling code generally. Of course, `G33` is brand new and I've never seen it in action once. No demonstration of any kind. I get a lot of assertions about a lot of things in Github-land, but as my writing teachers always said, "show me don't tell me." I'm not exactly thick-headed, but I do need to "see" what's being talked about to fully understand it. And today, with all our camera-toasters and laser-microphones there's no reason we shouldn't be able to demo stuff. When I see things actually working, the dawn suddenly appears.

Now, I'm finally down to under 100 issues, and I don't think I can get it much lower than 50 at any given time, so this is as good as it's going to get right now for me to be able to focus on the question of how best to let everyone adjust their P's and Z's. I want to release 1.1.9 in the next few days, so there's not a lot of time, but I like your PR and I want it to be included.

I very much have to sleep now, but I will look for you later.
